pkg/clustervalidator/clustervalidator.go

```diff
@@ -0,0 +1,84 @@
+package clustervalidator
```

Comment:
We should refrain from creating packages for types. Is there a reason this can't live in the  `Cluster` package? I would think things consuming the `cluster.Spec` would want to consume `cluster.Validator`.